go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -270,6 +296,32 @@ func (n *NodeController) DeleteNode(node *kapi.Node) error {
 	}
 
 	n.deleteFlowsByCookie(nameToCookie(node.Name))
+
+	cidr, _, _, err := getNodeDetails(node)
+	if cidr == nil || err != nil {
+		return fmt.Errorf("failed to lookup hybrid overlay node cidr for node %s: %v", node.Name, err)
+	}
+	if len(config.HybridOverlay.ClusterSubnets) == 0 {
+		// No static cluster subnet is provided in config. Try to detect the hybrid overlay node subnet dynamically
+		// Add a route via the hybrid overlay port IP through the management port
+		// interface for each hybrid overlay cluster subnet
+		mgmtPortLink, err := netlink.LinkByName(types.K8sMgmtIntfName)
```

Comment:
this looks like the same code you added above, probably better to move to a function

Reply:
Here is the deleting of the static routes, the above is adding.